src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let clone = Object.assign({}, state);
+
+  for (const item of actions) {
+    if (item.type === 'addProperties') {
+      Object.assign(clone, item.extraData);
+
+      const newData = Object.assign({}, clone);
+
+      history.push(newData);
```

Comment:
[DONT REPEAT YOURSELF] - You are creating a new object copy inside each if block. Instead, you should create a copy of the object once after all cases have been handled.